Invalid WARNING: Closing open connection prior to commit

**Describe the bug**
Logs are showing an invalid error when doing user local transaction management and committing a transaction. Does not happen in 1.2.0.CR1 but does on 999-SNAPSHOT
2020-01-24 16:09:46,216 WARN  [io.agr.pool] (executor-thread-27) Datasource '<default>': Closing open connection prior to commit


**Expected behavior**
This should not be logged

**Actual behavior**
Logs show:
2020-01-24 16:09:46,216 WARN  [io.agr.pool] (executor-thread-27) Datasource '<default>': Closing open connection prior to commit


**To Reproduce**
Steps to reproduce the behavior:

1) Run an app that uses resource local transactions. E.g.

@Path("/hello1")
@RequestScoped
public class GreetingResource1 {

    @Inject
    EntityManagerFactory myEmf;
    
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        
        EntityManager em = myEmf.createEntityManager();
        em.getTransaction().begin();
        em.createNativeQuery("SELECT 1").getResultList();
        em.getTransaction().commit();
        em.close();
        
        return "hello1";
    }
}
2) Call the rest endpoint and the logs will show the warnng when the line "em.getTransaction().commit();" is called


**Configuration**
```properties
# Add your application.properties here, if applicable.
Just standard props file to connect to a DB

**Screenshots**
NA

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux PCBLaptop 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "11.0.5" 2019-10-15
 - GraalVM version (if different from Java): NA
 - Quarkus version or git rev: 999-SNAPSHOT

**Additional context**
I only noticed this today. Yesterday it was fine

@AmsterdamFilho I have a test failure with your POC but no warnings.

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 7.81 s <<< FAILURE! - in br.com.quarkus.poc.AnimalResourceTest
[ERROR] testPut  Time elapsed: 1.413 s  <<< ERROR!
java.lang.RuntimeException: org.opentest4j.AssertionFailedError: expected: <dog> but was: <dawg>
	at br.com.quarkus.poc.AnimalResourceTest.testPut(AnimalResourceTest.java:36)
```

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/FastBootMetadataBuilder.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/FastBootMetadataBuilder.java
@@ -70,7 +70,6 @@ import org.hibernate.jpa.boot.spi.TypeContributorList;
 import org.hibernate.jpa.internal.util.LogHelper;
 import org.hibernate.jpa.internal.util.PersistenceUnitTransactionTypeHelper;
 import org.hibernate.jpa.spi.IdentifierGeneratorStrategyProvider;
-import org.hibernate.resource.jdbc.spi.PhysicalConnectionHandlingMode;
 import org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorBuilderImpl;
 import org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorBuilderImpl;
 import org.hibernate.service.Service;
@@ -268,22 +267,6 @@ public class FastBootMetadataBuilder {
         }
         //Agroal already does disable auto-commit, so Hibernate ORM should trust that:
         cfg.put(AvailableSettings.CONNECTION_PROVIDER_DISABLES_AUTOCOMMIT, Boolean.TRUE.toString());
-
-        /**
-         * Set CONNECTION_HANDLING to DELAYED_ACQUISITION_AND_RELEASE_AFTER_TRANSACTION
-         * as it generally performs better, at no drawbacks.
-         * The reason it's not the default in Hibernate ORM is that some containers suspect it to leak connections
-         * when running in this mode, as they trace resource handling across different beans.
-         * 
-         * @see org.hibernate.resource.jdbc.spi.PhysicalConnectionHandlingMode
-         */
-        {
-            final Object explicitSetting = cfg.get(AvailableSettings.CONNECTION_HANDLING);
-            if (explicitSetting == null) {
-                cfg.put(AvailableSettings.CONNECTION_HANDLING,
-                        PhysicalConnectionHandlingMode.DELAYED_ACQUISITION_AND_RELEASE_AFTER_TRANSACTION);
-            }
-        }
 
         if (readBooleanConfigurationValue(cfg, WRAP_RESULT_SETS)) {
             LOG.warn("Wrapping result sets is not supported. Setting " + WRAP_RESULT_SETS + " to false.");

